patches/0060-Create-Logging-Middleware-for-Lambda-service-that-do.patch

```diff
@@ -0,0 +1,158 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Florian Stadler <florian@pulumi.com>
+Date: Fri, 31 May 2024 12:29:36 +0200
+Subject: [PATCH] Create Logging Middleware for Lambda service that does not
+ log the lambda code archive
+
+When creating lambda functions and directly uploading the code, then the whole archive
+is being logged as a base64 encoded string as part of the HTTP request logger.
+In order to do so, multiple copies of the body are created in memory, which leads
+to memory bloating.
+This change fixes that by redacting the body in the logs for the Create/Update Lambda
+calls.
+
+diff --git a/internal/service/lambda/request_response_logger.go b/internal/service/lambda/request_response_logger.go
+new file mode 100644
+index 0000000000..58563221ab
+--- /dev/null
++++ b/internal/service/lambda/request_response_logger.go
+@@ -0,0 +1,107 @@
++package lambda
++
++import (
++	"context"
++	"fmt"
++	"github.com/aws/aws-sdk-go-v2/aws"
++	awsmiddleware "github.com/aws/aws-sdk-go-v2/aws/middleware"
++	"github.com/aws/smithy-go/middleware"
++	smithyhttp "github.com/aws/smithy-go/transport/http"
++	"github.com/hashicorp/aws-sdk-go-base/v2/logging"
++	"io"
++	"net/http"
++	"strings"
++	"time"
++	_ "unsafe"
++)
++
++const (
++	lambdaCreateOperation             = "CreateFunction"
++	lambdaUpdateFunctionCodeOperation = "UpdateFunctionCode"
++)
++
++// Replaces the upstream logging middleware from https://github.com/hashicorp/aws-sdk-go-base/blob/main/logger.go#L107
++// We do not want to log the Lambda Archive that is part of the request body because this leads to bloating memory
++type wrappedRequestResponseLogger struct {
++	wrapped middleware.DeserializeMiddleware
++}
++
++// ID is the middleware identifier.
++func (r *wrappedRequestResponseLogger) ID() string {
++	return "PULUMI_AWS_RequestResponseLogger"
++}
++
++func NewWrappedRequestResponseLogger(wrapped middleware.DeserializeMiddleware) middleware.DeserializeMiddleware {
++	return &wrappedRequestResponseLogger{wrapped: wrapped}
++}
++
++//go:linkname decomposeHTTPResponse github.com/hashicorp/aws-sdk-go-base/v2.decomposeHTTPResponse
++func decomposeHTTPResponse(ctx context.Context, resp *http.Response, elapsed time.Duration) (map[string]any, error)
++
++func (r *wrappedRequestResponseLogger) HandleDeserialize(ctx context.Context, in middleware.DeserializeInput, next middleware.DeserializeHandler,
```

Comment:
Guessing this was inlined and edited from upstream ? I think that's great and worth doing, just would be helpful to indicate that in the comment, and highlight the edits, just in case. Since we forked it the upstream changes to this handler will not propagate anymore but that sounds ok.